classifier_and_summarizer_interface.py

```diff
@@ -40,20 +40,23 @@ def preprocess_contents(self):
     def get_stampified_content(self):
         ''' returns the list of stamp pages'''
         # pre-process the contents first
-        self.preprocess_contents()
+        self._preprocess_contents()
 
         # classify the page as stampifiable or not
-        self.classify()
+        self._classify()
```

Comment:
early return?
```
        self._classify()
        
        if not self.is_stampifiable:
            return {
            "is_stampifiable": self.is_stampifiable,
            "stamp_pages": None
            }

        self._summarize()

        # order the stamp pages
        self._order_stamp_pages()     
```

Reply:
Thanks for the suggestion! But since `self._summarize` is called only when its stampifiable - `stamp_pages` will retain it's default value which is `None` and i'd also have to end up writing the return statement twice - i'm not sure how it would be better? Any reason why an early return is preferred here?

Reply:
Why should we call _order_stamp_pages if it is not stampifiable?
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

Reply:
Ah yes. So i guess we'll call both summarize and order stamp pages only if it's stampifiable- would that be alright?

Reply:
Or actually on a second look - i see that the early return would be better. Thanks for the suggestion!

Reply:
Updated accordingly.